### DISCUSSION TITLE
Building DFTD4 with meson

### DISCUSSION THREAD

**Discussion by saintdav** [dftd4/dftd4#191] [Q&A]
Hi, I am trying to build DFD4 with meson, but when I run `meson setup _build`, I receive the following message `config/meson.build:128:4: ERROR: Fortran shared or static library 'lapack' not found`. I have seen in the file meson_options.txt that
```
option(
  'lapack',
  type: 'combo',
  value: 'auto',
  yield: true,
  choices: ['auto', 'mkl', 'mkl-rt', 'openblas', 'netlib', 'custom'],
  description : 'linear algebra backend',
)

option(
  'custom_libraries',
  type: 'array',
  value: [],
  yield: true,
  description: 'libraries to load for custom linear algebra backend',
```
Thus, I added the option `-Dlapack=custom`, after trying the others and trying to specify the lapack library using `-Dcustom_libraries=something`, but I do not know how to do it. The computer that I am using has the Cray LibSci module which contains the LAPACK netlib library. This is the first time that I use meson and I am stuck at this point.
Any help will be greatly appreciated.
Thank you so much!

**Comment by awvwgk** [comment]
Sorry I don't have much experience with Cray systems, maybe ask your local cluster support for help with selecting the right libraries for linking.

**Reply by saintdav** [reply]
Hi Sebastian,
Thank you for your swift response. I have been looking online how to use a custom library with meson to no avail. Do you know how to do this? I installed meson locally in my account, as it is not provided by my cluster and I am worried that the support will not have experience using meson.
Thank you so much again!

**Reply by awvwgk** [reply]
From my understanding this should be all handled by the Cray Compiler wrappers, so you could leave the custom libraries empty? If you know the names of the libraries to link, just put them in as a comma separated list.

There is also support for CMake available, if that works better with Cray platforms (I usually recommend to stay away from the CMake as it quickly becomes painful to use).

**Reply by saintdav** [reply]
I managed to compile dftd4 with meson using the mkl library, which is also available in my cluster. I then tried to build VASP with support for DFTD4. The VASP executable is definitely seeing the dftd4 library, but it cannot recognise the IVDW=13 INCAR tag. Are the instructions provided in the Readme.md file only suitable for VASP6? Thank you again for your support!

**Comment by saintdav** [comment]
Hi Sebastian,
I have received support from my local cluster, but we are still unable to build VASP5 with support for DFTD4. Will this only work with VASP6. The VASP wiki mentions that in one instance under the description of the IVDW tag.
Thank you so much!

**Reply by awvwgk** [reply]
I believe it was implemented in Vasp6 first.

**Reply by saintdav** [reply]
Thanks. I will have to wait until we upgrade our license.
